charts/newrelic-infra-operator/templates/deployment.yaml

```diff
@@ -30,24 +30,24 @@ spec:
       {{- end }}
       {{- if .Values.image.pullSecrets }}
       imagePullSecrets:
-        {{- toYaml .Values.image.pullSecrets | nindent 8 }}
+        {{- include "newrelic.common.images.renderPullSecrets" ( dict "pullSecrets" ( list .Values.image.pullSecrets ) "context" .) | nindent 8 }}
       {{- end }}
```

Comment:
There are a couple of functions from the common library that are not implemented yet like `dnsConfig` and `hostNetwork`.

Let's leave it for a follow up PR :)